File: benchmarks/suite5/powmod.c
#include <stdint.h>
#include <stdio.h>
#include "bench.h"

int main(void) {
  int64_t acc = 1;
  const int64_t base = 3;
  const int64_t n = 2500000;
  for (int64_t i = 0; i < n; i++) {
    acc = (acc * base) % 1000000007;
  }
  suite5_print_i64(acc);
  return 0;
}

File: benchmarks/suite5/powmod.zig
const bench = @import("bench.zig");
const clib = @cImport({
    @cInclude("stdio.h");
});

pub fn main() void {
    var acc: i64 = 1;
    const base: i64 = 3;
    const n: i64 = bench.opaqueI64(2500000);
    var i: i64 = 0;
    while (i < n) : (i += 1) {
        acc = @rem(acc * base, 1000000007);
    }
    bench.printI64(acc);
}
